sources/libClangSharp/ClangSharp.h

```diff
@@ -12,7 +14,7 @@
 #endif
 
 #ifdef _MSC_VER
-#ifdef CLANGSHARP_LINKAGE
+#ifdef clangsharp_LINKAGE
```

Comment:
Shouldn't this be [`libClangSharp_EXPORTS`](https://cmake.org/cmake/help/v3.18/prop_tgt/DEFINE_SYMBOL.html)?

([`libClangSharp` because that's what it's called on Windows builds and this is for Windows.](https://github.com/microsoft/ClangSharp/blob/a42910cfcfd009880c2f12808b51f2dca25ba1c3/sources/libClangSharp/CMakeLists.txt#L41)

Better yet, can we just add our own preprocessor define instead of relying on the magic one added by CMake?

Reply:
no, `libclang` just uses `CLANG_LINKAGE`

Reply:
I'm slightly confused now. Pretty sure `libclang` uses `CINDEX_LINKAGE` and that's not what this review comment is referring to.

I see you changed it back to `#ifdef CLANGSHARP_LINKAGE`, but I don't think that's right either.

```cpp
#ifdef CLANGSHARP_LINKAGE
#define CLANGSHARP_LINKAGE EXTERN_C __declspec(dllexport)
#else
#define CLANGSHARP_LINKAGE EXTERN_C __declspec(dllimport)
#endif
```

This can only ever do two things:

1) `CLANGSHARP_LINKAGE` will be defined twice, causing an C4005 warning.
2) `CLANGSHARP_LINKAGE` will defined as `EXTERN_C __declspec(dllimport)` (which isn't right when we're building a DLL.)

I feel like I'm missing something here since it shouldn't even build without `CLANGSHARP_LINKAGE` (and even with it defined there'll be a warning.) Maybe I'm blind, but it doesn't look like libClangSharp is even built by the build scripts at all. Are you just manually invoking cmake and creating the NuGet package for it manually?

Reply:
It looks to be "working" because `CLANGSHARP_LINKAGE` isn't defined and therefore it is always exported. It ultimately doesn't matter because no one is going to use this from C code anyways, so most of it doesn't apply.
I will, however, fix it to just always export with a note that no one should be using this from another C library.

Reply:
Sounds good